From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: FloralDroid <opensource@floraldroid.dev>
Date: Fri, 14 Aug 2026 02:18:00 +0800
Subject: [PATCH] sepolicy: expose Floral ABI view properties

Declare the generated public and loader ABI properties so init can publish
the configured values and framework processes can read them.
---
diff --git a/private/property_contexts b/private/property_contexts
index a51fa3a..2d8b4d5 100644
--- a/private/property_contexts
+++ b/private/property_contexts
@@ -721,6 +721,12 @@ ro.product.cpu.abi       u:object_r:build_prop:s0 exact string
 ro.product.cpu.abilist   u:object_r:build_prop:s0 exact string
 ro.product.cpu.abilist32 u:object_r:build_prop:s0 exact string
 ro.product.cpu.abilist64 u:object_r:build_prop:s0 exact string
+ro.floral.nb.pubabi       u:object_r:build_prop:s0 exact string
+ro.floral.nb.pubabi32     u:object_r:build_prop:s0 exact string
+ro.floral.nb.pubabi64     u:object_r:build_prop:s0 exact string
+ro.floral.nb.loadabi      u:object_r:system_prop:s0 exact string
+ro.floral.nb.loadabi32    u:object_r:system_prop:s0 exact string
+ro.floral.nb.loadabi64    u:object_r:system_prop:s0 exact string
 
 ro.product.system.brand        u:object_r:build_prop:s0 exact string
 ro.product.system.device       u:object_r:build_prop:s0 exact string
